vms/vms/middleware.py

```diff
@@ -0,0 +1,11 @@
+import pytz
```

Comment:
You need to use `tzlocal`. Use `get_localzone`. To get now for the user you can use something like,

`now = timezone.now().astimezone(get_localzone())`. We need to make it work for local timezone of the user. pytz tzinfo's object corresponding to the local timezone. It works even during daylight savings time transitions when local time may be ambiguous. `dateutil.tz.tzlocal()` was broken, you need to check if it still is. Please let me know if you have any questions.

Reply:
@tapasweni-pathak I did some searching on tzlocal and understood about how to make it work for local timezone of the user. However, I did not find much related to its DST support. Also, based on the little that I learned, if DST is used, something like `pst.localize(datetime(2017, 10, 1)).dst()` `dst` information will not be manipulated properly and require naive-offset datetime which would generate errors because the former one will use offset-aware datetime. [Reference](https://medium.com/@nqbao/python-timezone-and-daylight-savings-e511a0093d0)

Reply:
Also this might sound stupid, but using tzlocal I should convert all the date and time values to local timezone separately inside the views or is there a way to automate this?

Reply:
> using tzlocal I should convert all the date and time values to local timezone separately inside the views or is there a way to automate this

Yes, perfect, one way is to have a view which only does this for us.

Reply:
Okay i will do that asap. Thanks :)